src/cmd/scan.go

```diff
@@ -4,50 +4,532 @@
+	if err != nil {
+		return "", err
+	}
+
+	baseScan.Metadata.Component.Name = compare.TrimDockerRegistryPrefixes(baseScan.Metadata.Component.Name)
+	newScan.Metadata.Component.Name = compare.TrimDockerRegistryPrefixes(newScan.Metadata.Component.Name)
+
+	if baseScan.Metadata.Component.Name != newScan.Metadata.Component.Name {
+		if !allowDifferentImages {
+			return "", fmt.Errorf("these scans are not for the same image: %s != %s", baseScan.Metadata.Component.Name, newScan.Metadata.Component.Name)
+		} else {
+			_, _ = fmt.Fprintf(os.Stderr, "Warning: these scans are not for the same image: %s != %s\n", baseScan.Metadata.Component.Name, newScan.Metadata.Component.Name)
 		}
+	}
+
+	vulnStatus := compare.GenerateComparisonMap(baseScan, newScan)
+
+	markdownTable, err := compare.GenerateComparisonMarkdown(baseScan, newScan, vulnStatus)
+	if err != nil {
+		return "", err
+	}
+	return markdownTable, nil
+}
 
-		vulnStatus := compare.GenerateComparisonMap(baseScan, newScan)
+var scanAndCompareCmd = &cobra.Command{
+	Use: "scan-and-compare",
+	Short: "Scan the current Zarf package, scan the last released package, and compare the scans." +
+		"This command is a combination of `scan`, `scan-released`, and `compare-scans`.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if outputDirectory == "" {
+			var err error
+			outputDirectory, err = os.MkdirTemp("", "scans")
+			logger.Info("Output directory", slog.String("dir", outputDirectory))
+			if err != nil {
+				return err
+			}
+		}
+		zarfYamlScanOutDir := outputDirectory + string(os.PathSeparator) + "zarfYaml"
+		logger.Debug("Scanning zarf.yaml images")
+		zarfYamlScanResults, err := scanZarfYamlImages(zarfYamlScanOutDir)
+		if err != nil {
+			return err
+		}
 
-		markdownTable, err := compare.GenerateComparisonMarkdown(baseScan, newScan, vulnStatus)
+		releasedScanOutDir := outputDirectory + string(os.PathSeparator) + "released"
+		releasedScanResults, err := scanReleased(releasedScanOutDir)
 		if err != nil {
 			return err
 		}
+		logger.Debug("Comparing scans", slog.Any("current", zarfYamlScanResults), slog.Any("released", releasedScanResults))
 
-		fmt.Println(markdownTable)
+		var builder strings.Builder
+
+		for flavor, flavorResults := range zarfYamlScanResults {
+			logger.Debug("Scanning flavor", slog.String("flavor", flavor))
+			if releasedFlavorResults, found := releasedScanResults[flavor]; found {
+				for key, scanFile := range flavorResults {
+					imageName := extractImageName(key)
+					for _, override := range imageNameOverrides {
+						parts := strings.SplitN(override, "=", 2)
+						if len(parts) == 2 && parts[1] == imageName {
+							imageName = parts[0]
+							logger.Debug("Found override image name for image", slog.String("image", key), slog.String("override", imageName))
+							break
+						}
+					}
+					if releasedScanFile, err := findMatchingScan(imageName, releasedFlavorResults, logger); err != nil {
+						return err
+					} else {
```

Comment:
There's no need for the else case here, you can safely assume that if there's no error it can continue, so:
```golang
if releasedScanFile, err := findMatchingScan(imageName, releasedFlavorResults, logger); err != nil {
	return err
}
// the rest of the code
```
reads so much easier :wink:

Reply:
without `else` it has to look like this:
```
releasedScanFile, err := findMatchingScan(imageName, releasedFlavorResults, logger)
if err != nil {
	return err
}
...
```

otherwise `releasedScanFile` is not visible in the code below. Coming from less verbose languages, I feel bad about assignments first and then `if err !=nil `'s everywhere. I guess I have to make peace with that...